app/Model/Attribute.php

```diff
@@ -3526,4 +3513,66 @@ private function generateTypeDefinitions()
             //'url-regex' => array('desc' => '', 'default_category' => 'Person', 'to_ids' => 0),
         );
     }
+
+    private function findAttributeByValue($value)
+    {
+        $type = $this->data['Attribute']['type'];
+        $conditions = [
+            'Attribute.event_id' => $this->data['Attribute']['event_id'],
+            'Attribute.type' => $type,
+            'Attribute.deleted' => 0,
+            'Attribute.object_id' => 0,
+        ];
+
+        if (isset($this->data['Attribute']['category'])) {
+            $conditions['Attribute.category'] = $this->data['Attribute']['category'];
+        }
+
+        if (in_array($type, $this->getCompositeTypes(), true)) {
+            $value = explode('|', $value);
+            $conditions['Attribute.value1'] = $value[0];
+            $conditions['Attribute.value2'] = $value[1];
+        } else {
+            $conditions['Attribute.value1'] = $value;
+        }
+
+        if (isset($this->data['Attribute']['id'])) {
+            $conditions['Attribute.id !='] = $this->data['Attribute']['id'];
+        }
+
+        return $this->find('first', [
+            'recursive' => -1,
+            'conditions' => $conditions,
+            'fields' => ['Attribute.id', 'Attribute.uuid']
+        ]);
+    }
+
+    /**
+     * If there is no id set but attribute value is set, check if the attribute already exists (by value) 
+     *
+     * {@inheritdoc }
+     *
+     * @param int|string $id ID of record to check for existence
+     * @return bool True if such a record exists
+     */
+    public function exists($id = null)
+    {
+        if (!isset($this->data['Attribute']['breakOnDuplicate']) || $this->data['Attribute']['breakOnDuplicate'] !== false) {
+            return parent::exists($id);
+        }
+
+        // find existing attribute by value
+        if (empty($id) && isset($this->data['Attribute']['value'])) {
+            $existingAttribute = $this->findAttributeByValue($this->data['Attribute']['value']);
+            if (!empty($existingAttribute)) {
+                $this->id = $existingAttribute['Attribute']['id'];
+                $this->data['Attribute']['id'] = $existingAttribute['Attribute']['id'];
+                $this->data['Attribute']['uuid'] = $existingAttribute['Attribute']['uuid'];
```

Comment:
1. Does it still work if `timestamp`  is not set?
2. Are there other functions using `exists` that would have their behavior changed due to the override (couldn't find any this question about the override)?

Reply:
1. It sets a new  `timestamp`, if it's present it sets the provided value, no changes in behavior here.
2. Direct calls in `Attribute::restore()`, but `$this->id` is set so the new code won't execute (regardless the `breakOnDuplicate` value). By the framework in `Model::_doSave()` and `Model::delete()`.